### DISCUSSION TITLE
Resize jpeg image to a specific size

### DISCUSSION THREAD

**Discussion by blakemcbride** [ImageMagick/ImageMagick#3488] [Help]
Greetings,

I have a large number of images at widely varying disk sizes.  I want to convert them to approximately 200K each.  That should be good enough quality for my purposes.  So, I specifically do not want a quality factor or reduction factor.  I want to be able to take in a 300K to 20M (on disk) jpeg file and covert it to 200K.  I understand I'll lose detail.  That's okay.  Thanks!

**Comment by fmw42** [comment]
For ImageMagick 7, use

`-define jpeg:extent={size}`

as described at https://legacy.imagemagick.org/Usage/formats/#jpg_write

`magick input.suffix -define jpeg:extent=200KB output.jpg`

For ImageMagick 6, change magick to convert

**Comment by blakemcbride** [comment]
I'm looking for a way to do it from Java though.   Any ideas?

**Reply by blakemcbride** [reply]
I wrote something that just calls the ImagicMagick command-line utility as a separate process from Java.  Simple.  Works fine.
